### PR TITLE
Add RTF_*, RTA_*, RTAX_*, RTM_* definitions on BSDs

### LINKED ISSUE #3711: Missing RTF_*, RTA_*, RTAX_*, RTM_* definitions on BSDs
Target: `x86_64-unknown-dragonfly`, `x86_64-unknown-freebsd`, `x86_64-unknown-netbsd`, `x86_64-unknown-openbsd`
API: [BSD 4.3 Reno](https://github.com/dspinellis/unix-history-repo/blob/BSD-4_3_Reno/usr/src/sys/net/route.h) (historic), [DragonFly](https://github.com/DragonFlyBSD/DragonFlyBSD/blob/master/sys/net/route.h), [FreeBSD](https://github.com/freebsd/freebsd-src/blob/main/sys/net/route.h), [NetBSD](https://github.com/netbsd/src/blob/trunk/sys/net/route.h), [OpenBSD](https://github.com/openbsd/src/blob/master/sys/net/route.h)
Test case: https://github.com/n0-computer/iroh/issues/2306
See also: [downstream fix](https://github.com/freebsd/freebsd-ports/blob/78ad38f04f51/games/jumpy/files/patch-libc), copy-pasted from macOS and likely incorrect/incomplete

### PR DESCRIPTION
**Pull request by tesuji**
Grep from:

* https://github.com/freebsd/freebsd-src/raw/main/sys/net/route.h
* https://github.com/DragonFlyBSD/DragonFlyBSD/raw/master/sys/net/route.h
* https://github.com/NetBSD/src/raw/trunk/sys/net/route.h
* https://github.com/openbsd/src/raw/master/sys/net/route.h

Closes #3711.

@rustbot review

### CODE REVIEW

File: src/unix/bsd/freebsdlike/dragonfly/mod.rs [hunk 1]
```diff
@@ -1522,6 +1522,18 @@ pub const NGROUPS: usize = 16;
 pub const RB_PAUSE: ::c_int = 0x40000;
 pub const RB_VIDEO: ::c_int = 0x20000000;
 
+// net/route.h
+pub const RTF_CLONING: ::c_int = 0x100;
+pub const RTF_PRCLONING: ::c_int = 0x10000;
+pub const RTF_WASCLONED: ::c_int = 0x20000;
+pub const RTF_MPLSOPS: ::c_int = 0x1000000;
+
+pub const RTM_VERSION: ::c_int = 7;
```
Comment (tesuji) [inline src/unix/bsd/freebsdlike/dragonfly/mod.rs:1531]: shall I remove this place holder?

Comment (jbeich) [reply src/unix/bsd/freebsdlike/dragonfly/mod.rs:1531]: Would be inconsistent with macOS:
https://github.com/rust-lang/libc/blob/acc7bb1a38e2b9d6a18f28279d80d464c4c25089/src/unix/bsd/apple/mod.rs#L4770

Besides, RTM_VERSION is [used by iroh-net](https://github.com/n0-computer/iroh/blob/370075c6d5689ad4349664eb9b51ec0a5e7e4e81/iroh-net/src/net/interfaces/bsd.rs#L371).

File: src/unix/bsd/freebsdlike/freebsd/mod.rs [hunk 2]
```diff
@@ -4559,10 +4559,6 @@ pub const CPU_WHICH_JAIL: ::c_int = 5;
 pub const RTF_LLDATA: ::c_int = 0x400;
 pub const RTF_FIXEDMTU: ::c_int = 0x80000;
 
-pub const RTM_VERSION: ::c_int = 5;
```
Comment (jbeich) [inline src/unix/bsd/freebsdlike/freebsd/mod.rs]: Contradicts the commit message. Also, inconsistent with macOS, DragonFly, NetBSD, OpenBSD.

**Comment by jbeich** [comment]
Do you plan to unify with macOS by moving RTM_VERSION and some other constants to src/unix/bsd/mod.rs ?

**Comment by jbeich** [comment]
Please, don't remove *_MAX if you only going to keep it for macOS. It's actually used in the wild:
```rust
error[E0425]: cannot find value `RTAX_MAX` in crate `libc`
   --> cargo-crates/iroh-net-0.16.2/src/net/interfaces/bsd.rs:875:46
    |
875 |     let mut addrs = Vec::with_capacity(libc::RTAX_MAX as usize);
    |                                              ^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `RTAX_MAX` in crate `libc`
   --> cargo-crates/iroh-net-0.16.2/src/net/interfaces/bsd.rs:879:23
    |
879 |     for i in 0..libc::RTAX_MAX as usize {
    |                       ^^^^^^^^ not found in `libc`
    |
help: you might have meant to write `.` instead of `..`
    |
879 -     for i in 0..libc::RTAX_MAX as usize {
879 +     for i in 0.libc::RTAX_MAX as usize {
    |
```

**Comment by tgross35** [comment]
At first glance this looks okay, but there are a couple things needed:

1. Target `main`
2. Squash the commits
3. Update the `libc-test/semver` files since this now affects more platforms

**Comment by larseggert** [comment]
It'd be great to land this, so https://github.com/mozilla/mtu/pull/29 would work on the BSDs.

**Comment by tgross35** [comment]
Going to wait on the backport until https://github.com/rust-lang/libc/pull/3970 is figured out.

**Pull request merged**